File: android/src/tiramisu/java/de/idrinth/habitevaluator/android/NotificationHelper.java
package de.idrinth.habitevaluator.android;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;

/**
 * Tiramisu flavor notification helper (API 33+).
 * NotificationChannel is created unconditionally and the
 * POST_NOTIFICATIONS permission is always handled at runtime.
 */
public final class NotificationHelper {

    public static final String CHANNEL_ID = "habit_evaluator_reminders";

    private NotificationHelper() {
    }

    public static void ensureNotificationChannel(Context context) {
        NotificationChannel channel = new NotificationChannel(
                CHANNEL_ID,
                context.getString(R.string.reminder_channel_name),
                NotificationManager.IMPORTANCE_DEFAULT);
        channel.setDescription(context.getString(R.string.reminder_channel_description));
        NotificationManager manager = (NotificationManager)
                context.getSystemService(Context.NOTIFICATION_SERVICE);
        if (manager != null) {
            manager.createNotificationChannel(channel);
        }
    }

    public static NotificationPermissionHandler permissionHandler() {
        return new Api33PermissionHandler();
    }
}

File: android/src/tiramisu/kotlin/de/idrinth/habitevaluator/android/NotificationHelper.kt
package de.idrinth.habitevaluator.android

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context

object NotificationHelper {
    const val CHANNEL_ID = "habit_evaluator_reminders"

    fun ensureNotificationChannel(context: Context) {
        val channel = NotificationChannel(
            CHANNEL_ID,
            context.getString(R.string.reminder_channel_name),
            NotificationManager.IMPORTANCE_DEFAULT
        ).apply {
            description = context.getString(R.string.reminder_channel_description)
        }
        val manager = context.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager
        manager?.createNotificationChannel(channel)
    }

    fun permissionHandler(): NotificationPermissionHandler = Api33PermissionHandler()
}
